packages/gatsby/package.json

```diff
@@ -147,8 +147,10 @@
   },
   "files": [
     "./cache-dir",
-    "./dist"
+    "./dist",
+    "./index.d.ts"
   ],
+  "types": "index.d.ts",
```

Comment:
Is this required? As far as I can tell it's not necessary if `index.d.ts` is in the package root, [the TypeScript docs say "it is advisable to do so"](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html), but don't mention why it's advisable.

Reply:
By default, the TS compiler will check for either the `types` or `typings` key in a module's `package.json` and will include their declarations, statement completions and such if found.

If you write your project with TS and set it to automatically output declaration files (`"declaration": true`), you might not need to do this, because of how TS handles `.d.ts` files with the same name as their parent module. But since we're including typings manually, it *is* highly recommended to include it in `package.json`.